src/mod/asr_tts/mod_unimrcp/mod_unimrcp.c

```diff
@@ -3820,12 +3835,14 @@ static apt_bool_t recog_on_message_receive(mrcp_application_t *application, mrcp
 					recog_channel_set_result_headers(schannel, recog_hdr);
 					recog_channel_set_results(schannel, result);
 				}
+				prometheus_increment_asr_success();
 			} else {
 				char *completion_cause = switch_mprintf("Completion-Cause: %03d", recog_hdr->completion_cause);
 				switch_log_printf(SWITCH_CHANNEL_UUID_LOG(schannel->session_uuid), SWITCH_LOG_DEBUG, "(%s) No result\n", schannel->name);
 				recog_channel_set_result_headers(schannel, recog_hdr);
 				recog_channel_set_results(schannel, completion_cause);
 				switch_safe_free(completion_cause);
+				prometheus_increment_asr_failure();
```

Comment:
I don't think we need to increment failure here. Similar to how we handle dialplan responses, we don't need to increment failure when dialplan responded with a sip error 4xx. In that perspective the request was process and we got a response which is a success result.

Reply:
Good point. Updated.